packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,21 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, options = {}) => {
+const matchPath = (pathname, options = {}, parent) => {
   if (typeof options === 'string')
     options = { path: options }
 
-  const { path = '/', exact = false, strict = false } = options
+  const { path, exact = false, strict = false } = options
+
+  if (path == null)
+    return parent != null
+      ? parent
+      : {
+          url: '/_FAKE_MATCH_DO_NOT_ATTEMPT_TO_RESOLVE_USING_THIS_OR_YOU_WILL_BE_DISAPPOINTED',
```

Comment:
Okay, I actually did this. So far, the only use for `match.url` that I have seen* has been with concatenating paths using the parent route. When the parent match is `null`, we cannot provide an accurate parent `match.url`. Having the error explicitly defined in the URL like this will make it more obvious to anyone who attempts to resolve using it that something is not right with what they are trying to do.

This isn't a make or break change for me, but I think that if we try to use some other value like root or the current URL, we will end up with a lot of people asking why they are getting the wrong URL when they attempt to resolve.

*It can actually be useful as a transition key for animated switches, but it isn't actually accessible with the `<Switch>` component, so I'm ignoring that here.